sa/saro.go

```diff
@@ -736,21 +736,21 @@ func (ssa *SQLStorageAuthorityRO) GetOrder(ctx context.Context, req *sapb.OrderR
 		}
 		order.V2Authorizations = v2AuthzIDs
 
-		names, err := namesForOrder(ctx, tx, order.Id)
+		// Get the partial Authorization objects for the order
+		authzValidityInfo, err := getAuthorizationStatuses(ctx, tx, order.V2Authorizations)
+		// If there was an error getting the authorizations, return it immediately
 		if err != nil {
-			return nil, err
+			return "", err
```

Comment:
This should remain `nil`, as line 783 is expecting the result from this transaction to be a `*corepb.Order`.
```suggestion
			return nil, err
```

Reply:
Why doesn't the compiler/vet complain about returning an untyped string constant on NewOrderAndAuthzs when the return values must be `*corepb.Order, error`?

I am trying to figure this out myself and got [this far](https://go.dev/play/p/DcHOMH9T4xH).

Reply:
Because we're inside a [transaction's anonymous function](https://github.com/letsencrypt/boulder/blob/581240ea9a0d79114d5e11c59e828f80be16c9bd/sa/saro.go#L700), which has return values of `interface{}, error`.

Reply:
Terrifying, thank you.